### ISSUE TITLE
Refactor tests to have shared setup logic

### ISSUE THREAD

**Issue by aaronpowell** [Azure/data-api-builder#1546]
With the changes in #1402 there is repeated code for how we setup the mock file system environment across the CLI tests.

We should refactor the repeated code into a helper method so that we are more consistent in how we setup the tests.

## Context

Yep, I'll make a separate task for that as I think there's a bit of "general cleanup" we can do once this is merged to have some more shared code and things like that, but it's more "nice to have" than "must have".

_Originally posted by @aaronpowell in https://github.com/Azure/data-api-builder/pull/1402#discussion_r1239136401_
